handler/transport/backend.go

```diff
@@ -115,7 +117,9 @@ func (b *Backend) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 
-	setGzipReader(beresp)
+	if err = setGzipReader(beresp); err != nil {
+		b.upstreamLog.LogEntry().WithContext(req.Context()).WithError(err).Error()
```

Comment:
return/exit the method?

Reply:
since `setGzipReader` resets the body on error the response can still be processed